pkg/controller/controller_utils.go

```diff
@@ -629,6 +670,86 @@ func maxContainerRestarts(pod *api.Pod) int {
 	return maxRestarts
 }
 
+// ControlActivePods first filters out inactive pods, then it tries to take
+// control of the remaining pods that matching the selector. It returns the pods
+// that are active and are controlled by the controller, and the pods that
+// should be abandoned by the controller.
+func ControlActivePods(podControl PodControlInterface,
+	controllerObject api.ObjectMeta,
+	controllerSelector labels.Selector,
+	controllerKind unversioned.GroupVersionKind,
+	pods []api.Pod,
+) (controlledPods []*api.Pod, toAbandonPods []*api.Pod) {
+	for i := range pods {
+		pod := pods[i]
+		if !IsPodActive(pod) {
+			glog.V(4).Infof("Ignoring inactive pod %v/%v in state %v, deletion time %v",
+				pod.Namespace, pod.Name, pod.Status.Phase, pod.DeletionTimestamp)
+			continue
+		}
+		controlled, controllerRef := ControllerOf(pod.ObjectMeta)
+		if controlled {
+			if controllerRef.UID == controllerObject.UID {
+				// already controlled
+				if controllerSelector.Matches(labels.Set(pod.Labels)) {
+					controlledPods = append(controlledPods, &pod)
+				} else {
+					toAbandonPods = append(toAbandonPods, &pod)
+				}
+			} else {
+				// ignoring the pod controlled by other controller
+				glog.V(4).Infof("Ignoring pod %v/%v, it's owned by [%s/%s, name: %s, uid: %s]",
+					pod.Namespace, pod.Name, controllerRef.APIVersion, controllerRef.Kind, controllerRef.Name, controllerRef.UID)
+				continue
+			}
+		} else {
+			if !controllerSelector.Matches(labels.Set(pod.Labels)) {
+				continue
+			}
+			// try to take control
+			if ControlPod(podControl, controllerObject, controllerKind, pod) {
+				// successfully take control
+				controlledPods = append(controlledPods, &pod)
+			} else {
+				// failed to take control, ignore the pod
+				continue
+			}
+		}
+	}
+	return controlledPods, toAbandonPods
+}
+
+// ControllerOf returns if the controllee is controlled by a controller and if
+// so, a reference to the controller.
+func ControllerOf(controllee api.ObjectMeta) (controlled bool, controllerRef api.OwnerReference) {
```

Comment:
can you rename to something like getParentController?
Also can you pass back a pointer and use the nil pointer instaed of a bool? 

nit: why is this method public?